orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,265 @@
+from typing import Optional
+
+from AnyQt.QtCore import Qt, QAbstractTableModel, QSize, QItemSelectionModel, \
+                         QItemSelection
+from AnyQt.QtWidgets import QSizePolicy, QApplication, QTableView
+from AnyQt.QtGui import QBrush, QColor
+
+from Orange.data import Table
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+from nltk import ConcordanceIndex
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.preprocess import Preprocessor, WordPunctTokenizer
+
+
+class DocumentSelectionModel(QItemSelectionModel):
+    """Sets selection for QTableView. Creates a set of selected documents."""
+
+    def select(self, selection, flags):
+        # which rows have been selected
+        indexes = selection.indexes() if isinstance(selection, QItemSelection) \
+                  else [selection]
+        # indexes[0].row() == -1 indicates clicking outside of the table
+        if len(indexes) == 1 and indexes[0].row() == -1:
+            self.clear()
+            return
+        word_index = self.model().word_index
+        selected_docs = {word_index[index.row()][0] for index in indexes}
+        selected_rows = [
+            row_index for row_index, (doc_index, _) in enumerate(word_index)
+            if doc_index in selected_docs]
+        selection = QItemSelection()
+        # select all rows belonging to the selected document
+        for row in selected_rows:
+            index = self.model().index(row, 0)
+            selection.select(index, index)
+        super().select(selection, flags)
+
+
+class ConcordanceModel(QAbstractTableModel):
+    "A model for constructing concordances from text."
+
+    def __init__(self):
+        QAbstractTableModel.__init__(self)
+        self.word = None
+        self.corpus = None
+        self.indices = None
+        self.word_index = None
+        self.width = 8
+
+    def set_word(self, word):
+        self.modelAboutToBeReset.emit()
+        self.word = word
+        if self.indices is None:
+            self.word_index = None
+        else:
+            self.word_index = [
+                (doc_idx, offset) for doc_idx, doc in enumerate(self.indices)
+                 for offset in doc.offsets(self.word)]
```

Comment:
Remove one space. It's actually misleading; it looks as if `for` is inside the parentheses.